Mathlib/Computability/TuringMachine.lean

```diff
@@ -1079,6 +1079,12 @@ def step (M : Machine₀) : Cfg₀ → Option Cfg₀ :=
     | Stmt.write a => T.write a⟩
 #align turing.TM0.step Turing.TM0.step
 
+/-- Chainable step function -/
+def step_chain (M : Machine₀) (cfg : Option Cfg₀) : Option Cfg₀ := cfg.bind (step M)
+
+/-- Step a given number of times -/
+def multistep (M : Machine₀) (n : ℕ) (cfg : Cfg₀) : Option Cfg₀ := (step_chain M)^[n] (some cfg)
```

Comment:
It's not clear to me that these definitions are useful; or at least, they should be defined far more generally elsewhere as mechanisms to iterate a monad `A -> m A`.

Reply:
Yes, this could be generalized, but only to partial functions that may return “none”.

multistep can be used to defined polynomial time computations. It could be used to show that this concept is invariant across the machine types because the emulators are poly time. It might also be used to show that the composition of two poly time functions is poly time

Reply:
To be clear, I mean something like:

```lean 4
def Nat.iterateM {m} [Monad m] (f : α → m α) (n : ℕ) (a : α) : m α :=
   (. >>= f)^[n] (pure a)
```
(untested)

Reply:
That seems like `Nat.iterate` and `Computation.runFor`. The Computation and TuringMachine parts of mathlib seem unconnected. 

To include the TM quintet definition in mathlib, should the PR meet some criterion, for instance, a theorem not just a def, say a TMQ emulator of TM0 and vice versa?